src/common/utils.cc

```diff
@@ -21,15 +21,26 @@ namespace knowhere {
 
 const float FloatAccuracy = 0.00001;
 
-void
+float
 NormalizeVec(float* x, int32_t d) {
     float norm_l2_sqr = faiss::fvec_norm_L2sqr(x, d);
     if (norm_l2_sqr > 0 && std::abs(1.0f - norm_l2_sqr) > FloatAccuracy) {
         float norm_l2 = std::sqrt(norm_l2_sqr);
         for (int32_t i = 0; i < d; i++) {
             x[i] = x[i] / norm_l2;
         }
+        return norm_l2;
+    }
+    return 1.0f;
+}
+
+std::vector<float>
+NormalizeVecs(float* x, size_t rows, int32_t dim) {
+    std::vector<float> normals;
```

Comment:
need do : normals.resize(rows) at the begenning

Reply:
norms as return value is more convenient because users doesn't need to transfer the norms vector if they do not care the norms value.